### PR TITLE
feat: hide pox-2.stack-increase

### PR DESCRIPTION
**Pull request by friedger**
This PR
* hides pox-2.stacks-increase calls in preparation for Stacks 2.2
  * in the sandbox's list of available functions
  * in the contract page's list of available functions
  * in the sandbox deep link http://localhost:3000/sandbox/contract-call/SP000000000000000000002Q6VF78.pox-2/stack-increase?chain=mainnet
  
image

How to test?
Check the list of available functions on the following pages:
* visit /txid/SP000000000000000000002Q6VF78.pox-2?chain=mainnet 
* visit sandbox/contract-call/SP000000000000000000002Q6VF78.pox-2?chain=mainnet
* visit sandbox/contract-call/SP000000000000000000002Q6VF78.pox-2/stack-increase?chain=mainnet

### CODE REVIEW

File: src/app/sandbox/components/ContractCall/AvailableFunctionsView.tsx [hunk 1]
```diff
@@ -67,14 +67,23 @@ export const AbiFunction = forwardRef<
...
+
 export const AvailableFunctionsView: FC<{
   contract: ContractWithParsedAbi;
   contractId: string;
 }> = ({ contractId, contract }) => (
   <Section overflowY="auto" flexGrow={1} title="Available functions">
     {contract?.abi?.functions.map(
       (abiFn: any) =>
-        abiFn.access !== 'private' && (
+        hideFn(contractId, abiFn) && (
```
Comment (diwakergupta) [inline src/app/sandbox/components/ContractCall/AvailableFunctionsView.tsx:86]: did you mean `showFn`?

Comment (friedger) [reply src/app/sandbox/components/ContractCall/AvailableFunctionsView.tsx]: fixed

File: src/app/common/utils/sandbox.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function showFn(contractId: string, abiFn: any) {
+  return (
```
Comment (diwakergupta) [inline src/app/common/utils/sandbox.ts:2]: Can you add a brief comment to document this function and in particular, make a note about stacks-increase in PoX-2.

Separately, I found this implementation a bit hard to reason through: basically we're saying show a function IF:
* it's NOT private, _and_
* it NOT stacks-increase in POX-2

IMO an implementation that uses fewer negatives might be easier to understand and maintain. Something like:

hide a function IF
* it's private OR
* it's stacks-increase in testent/mainnet pox-2

File: src/app/sandbox/components/ContractCall/AvailableFunctionsView.tsx [hunk 3]
```diff
@@ -10,7 +11,7 @@ import { Text } from '@/ui/typography';
 import { forwardRef, useColorMode } from '@chakra-ui/react';
 import { mdiApi, mdiFunction } from '@mdi/js';
 import Icon from '@mdi/react';
-import React, { FC } from 'react';
```
Comment (diwakergupta) [inline src/app/sandbox/components/ContractCall/AvailableFunctionsView.tsx:13]: This seems like an unrelated removal? Can we avoid any unnecessary changes in the PR to simplify the diff? Thanks.

**Review by diwakergupta — CHANGES_REQUESTED** [review]
I'm not super familiar with the explorer code base but overall changes LGTM. If you can add some docs to `showFn` we can get this merged. Thanks Friedger!

File: src/app/sandbox/components/ContractCall/AvailableFunctionsView.tsx (hunk 3 above)
Comment (friedger) [reply src/app/sandbox/components/ContractCall/AvailableFunctionsView.tsx:13]: this happened when I ran format

File: src/app/common/utils/sandbox.ts (hunk 2 above)
Comment (friedger) [reply src/app/common/utils/sandbox.ts:2]: My motivation was to have a cleaner code in the components:
`showFn(contractId, abiFn) && <FunctionView.../>`

and when using filter the abi

**Review by diwakergupta — APPROVED** [review]
LGTM!
